### DISCUSSION TITLE
Environment variable verification on startup

### DISCUSSION THREAD

**Discussion by roshan-sama** [blitz-js/blitz#2604] [RFCs]
With the new initServer feature in blitz.config.ts, I have begun to verify all my environment variables using a function that calls all the variables in .env

`blitz.config.ts`
```
const config: BlitzConfig = {
  ...
  experimental: {
      initServer() {
        verifyEnvs()
      },
    },
}
```

`integrations\envsafe.ts`
```
import { num, envsafe, str } from 'envsafe';

export const env = envsafe({
  NODE_ENV: str({
    devDefault: 'development',
    choices: ['development', 'test', 'production'],
  }),
  NEXT_PUBLIC_APP_DISPLAY_NAME: str(),
  DATABASE_URL: str(),
  NEXT_PUBLIC_GOOGLE_MAPS_API_KEY: str()
});

export const verifyEnvs = () => {
    const a = env.NODE_ENV
    const b = env.NEXT_PUBLIC_APP_DISPLAY_NAME
    const c = env.DATABASE_URL
    const d = env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY
    console.log("Verified environment variables")
}
```

Would it make sense to add something this by default to new apps? Looking at https://github.com/blitz-js/blitz/discussions/1152 it looks like there is an alternative to set client env vars apart from the NEXT_PUBLIC_ environment variables so that could potentially be implemented together with this

**Comment by flybayer** [comment]
Good suggestion! I *think* it probably would make sense, but I'm not super happy with envsafe. I'm currently using it, but it's tricky to separately validate client vs server env vars. 

Ideally the validation library would have built in support for isomorphic things. Ideally a way to define all client and server envs in the same file. Currently with envsafe it has to be separate files.
